ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -5,78 +5,130 @@
 
 import Component from '@glimmer/component';
 import { action } from '@ember/object';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
+import { service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { ValidationMap } from 'vault/vault/app-types';
+import errorMessage from 'vault/utils/error-message';
 
-import type SecretEngineModel from 'vault/models/secret-engine';
-import type { TtlEvent } from 'vault/app-types';
+import type LeaseConfigModel from 'vault/models/aws/lease-config';
+import type RootConfigModel from 'vault/models/aws/root-config';
+import type Router from '@ember/routing/router';
+import type Store from '@ember-data/store';
+import type FlashMessageService from 'vault/services/flash-messages';
 
 /**
- * @module ConfigureAwsComponent
+ * @module ConfigureAwsComponent is used to configure the AWS secret engine
+ * A user can configure the endpoint root/config and/or lease/config. 
+ * The fields for these endpoints are on one form.
  *
  * @example
  * ```js
  * <SecretEngine::ConfigureAws
-    @model={{model}}
-    @tab={{tab}}
-    @accessKey={{accessKey}}
-    @secretKey={{secretKey}}
-    @region={{region}}
-    @iamEndpoint={{iamEndpoint}}
-    @stsEndpoint={{stsEndpoint}}
-    @saveAWSRoot={{action "save" "saveAWSRoot"}}
-    @saveAWSLease={{action "save" "saveAWSLease"}} />
+    @rootConfig={{this.model.aws-root-config}}
+    @leaseConfig={{this.model.aws-lease-config}} 
+    @id={{this.model.id}} 
+    />
  * ```
  *
- * @param {object} model - aws secret engine model
- * @param {string} tab - current tab selection
- * @param {string} accessKey - AWS access key
- * @param {string} secretKey - AWS secret key
- * @param {string} region - AWS region
- * @param {string} iamEndpoint - IAM endpoint
- * @param {string} stsEndpoint - Sts endpoint
- * @param {Function} saveAWSRoot - parent action which saves AWS root credentials
- * @param {Function} saveAWSLease - parent action which updates AWS lease information
- *
+ * @param {object} rootConfig - AWS config/root model
+ * @param {object} leaseConfig - AWS config/lease model
+ * @param {string} id - name of the AWS secret engine, ex: 'aws-123'
```

Comment:
This should be the backend path right? I wonder if we should rename this to backendPath instead for clarity

Reply:
Good idea.